spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -194,9 +195,9 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
     Map<NamedReference, ColumnStatistics> colStatsMap = Collections.emptyMap();
     if (readConf.reportColumnStats() && cboEnabled) {
       colStatsMap = Maps.newHashMap();
-      List<StatisticsFile> files = table.statisticsFiles();
-      if (!files.isEmpty()) {
-        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
+      Optional<StatisticsFile> statisticsFile = statisticsFile(snapshot);
+      if (statisticsFile.isPresent()) {
+        List<BlobMetadata> metadataList = statisticsFile.get().blobMetadata();
```

Comment:
@karuppayya Sorry for missing this earlier, I think we may want to consider a table API for resolving a statistics file based on a snapshot, `statisticsFileFor`. The implementation of that API could just do a best effort search of the statistics file for a given snapshot, and if one cannot be found just return the most recent one. 

If an engine integration needs the exact statistics and the API response isn't it, that's OK since the engine can then just ignore the statistics file. But i think in the most common cases, having an out of date statistics file is probably acceptable and so the API should probably default to the best effort lookup.

This is analagous to what happens in view.dialectFor API where a best effort for a given dialect is searched but if one cannot be found the first representation is returned. Engines like Trino which require the strict dialect can use the response and compare against the desired and fail accordingly. Other engines like Spark don't do the strict lookup.

Reply:
+1 to introduce the table API for retrieving the stats.
But should we do a best effort here or jsut return empty when there arent stats for the snasphot?  
We dont have a means to compare against a baseline to figure if its an approximation, unlike dialects where it could be validated.

Reply:
Hi @karuppayya , i saw the latest changes, but still as per the latest changes it will take the latest Snapshot Id right, and it filters over it. Which means that if the Analyze procedure is not executed for the latest snapshot, it won't find the stat file right. Hence it is not doing the best effort search of the statistics file for a given snapshot right?

Instead, it should pick the last existing statistics file right, so that we may get some benefit out of it at least in the query planning. Could you please help me understand the current behavior.

Reply:
@jeesou Yes, the current code doesnt return any available stats.
I think returning a best effort stats can result in bad decisions by optimizer based on when the stats were computed.
We can introduce a config to let users decide if they are fine with best effort search. This way the user is also aware of it, instead of doing it transparently. WDYT? @jeesou @amogh-jahagirdar

Reply:
Yes @karuppayya , making it config based seems like a better idea, giving the user more control over it.